SparseVector/Task/SparseVector.cs

```diff
@@ -0,0 +1,223 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace SparseVector;
+
+/// <summary>
+/// Class that represents vector which contains many zeros.
+/// </summary>
+public class SparseVector
+{
+    private readonly int size = 0;
+
+    private Dictionary<int, int> vector = new ();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SparseVector"/> class.
+    /// </summary>
+    /// <param name="original">Original instance of sparse vector, from which size and elements will be copied.</param>
+    public SparseVector(SparseVector original)
+    {
+        this.size = original.Size;
+        this.vector = new (original.vector);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SparseVector"/> class.
+    /// </summary>
+    /// <param name="size">Size of the vector.</param>
+    public SparseVector(int size)
+    {
+        this.size = size;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SparseVector"/> class.
+    /// </summary>
+    /// <param name="vector">List, that contains pairs of (index, number) that is going to be contained in sparse vector.</param>
+    /// <param name="size">Size of the sparse vector.</param>
+    public SparseVector(List<(int, int)> vector, int size)
+    {
+        this.size = size;
+        foreach (var element in vector)
+        {
+            if (element.Item1 >= size)
+            {
+                throw new IndexOutOfRangeException();
+            }
+
+            if (element.Item2 == 0)
+            {
+                continue;
+            }
+
+            if (this.vector.ContainsKey(element.Item1))
+            {
+                throw new ArgumentException("Initializer vector contained same index more than once.");
+            }
+
+            this.vector.Add(element.Item1, element.Item2);
+        }
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SparseVector"/> class.
+    /// </summary>
+    /// <param name="givenVector">Vector with numbers that is going to be added to sparse vector.</param>
+    /// <param name="size">Size of the vector.</param>
+    /// <exception cref="IndexOutOfRangeException">Being thrown if given vector`s size is bigger than size of sparse vector. </exception>
+    public SparseVector(List<int> givenVector, int size)
+    {
+        if (givenVector.Count > size)
+        {
+            throw new IndexOutOfRangeException();
+        }
+
+        this.size = size;
+        for (int i = 0; i < givenVector.Count; i++)
+        {
+            if (givenVector[i] == 0)
+            {
+                continue;
+            }
+
+            this.vector.Add(i, givenVector[i]);
+        }
+    }
+
+    /// <summary>
+    /// Gets size of the vector.
+    /// </summary>
+    public int Size { get => this.size; }
```

Comment:
```suggestion
    public int Size => this.size;
```
Это то же самое, но короче